### ISSUE TITLE
async-std-1.6 freeze

### ISSUE THREAD

**Issue by hhggit** [async-rs/async-std#808]
The following code run OK on smol, and async-std-1.5, but freeze on async-std-1.6.

``` toml
[dependencies]
async-speed-limit = "0.3.0"
smol = {version = "0.1.11", optional=true}
async-std = {version = "=1.6.0", optional=true}
#async-std = {version = "=1.5.0", optional=true}
```

``` rust
use std::{
    future::Future,
    // io::{stdout, Write},
    // time::Instant,
};

fn main() {
    block_on(async {
        use async_speed_limit::limiter::Limiter;
        let limiter = <Limiter>::new(8.0 * 1024.0 * 1024.0);

        // let mut i = 0usize;
        // let t = Instant::now();
        loop {
            // if i % 64 == 0 {
            //     println!("\n{} {}ms", i, t.elapsed().as_millis());
            // }
            //
            // print!(".");
            // stdout().flush().unwrap();

            limiter.consume(1024).await;

            // i += 1;
        }
    });
}

#[cfg(feature = "smol")]
fn block_on(f: impl Future) {
    smol::run(f);
}

#[cfg(feature = "async-std")]
fn block_on(f: impl Future) {
    async_std::task::block_on(f);
}
```

**Comment by dignifiedquire** [comment]
How long did you leave it running for until it would freeze? I am unable to reproduce this on the latest version with smol@0.1.11

**Comment by hhggit** [comment]
Few seconds, and always reproduces. Did you run it in feature "async-std" ?

**Comment by dignifiedquire** [comment]
I did, with current master, and it worked without any issues

**Comment by hhggit** [comment]
Close as others could not reproduce this.

**Comment by hhggit** [comment]
@dignifiedquire This issue has been fixed in async-std-1.6.2, thanks for your great work!
